File: anime-social-platform/backend/src/main/java/com/animesocial/platform/model/Resource.java
package com.animesocial.platform.model;

import java.time.LocalDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import com.animesocial.platform.listener.ResourceEntityListener;

import lombok.Data;

/**
 * 资源实体类
 * 
 * 表示系统中的资源信息，包含：
 * 1. 基本信息：ID、标题、描述等
 * 2. 文件信息：文件URL、类型、大小等
 * 3. 统计信息：下载次数、收藏次数等
 * 4. 上传信息：上传用户、上传时间等
 */
@Data
@Entity
@Table(name = "resources")
@EntityListeners(ResourceEntityListener.class)
public class Resource {
    /**
     * 资源ID
     */
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    
    /**
     * 上传用户ID
     */
    @Column(name = "user_id", nullable = false)
    private Integer userId;
    
    /**
     * 资源标题
     */
    @Column(nullable = false)
    private String title;
    
    /**
     * 资源描述
     */
    @Column(columnDefinition = "TEXT")
    private String description;
    
    /**
     * 文件URL
     */
    @Column(name = "file_url", nullable = false)
    private String fileUrl;
    
    /**
     * 封面图片URL（可选）
     */
    @Column(name = "cover_url")
    private String coverUrl;
    
    /**
     * 文件类型
     */
    @Column(name = "file_type", nullable = false)
    private String fileType;
    
    /**
     * 文件大小（KB）
     */
    @Column(name = "file_size")
    private Integer fileSize;
    
    /**
     * 上传时间
     */
    @Column(name = "upload_time")
    private LocalDateTime uploadTime;
    
    /**
     * 下载次数
     */
    @Column(name = "download_count", columnDefinition = "INTEGER DEFAULT 0")
    private Integer downloadCount;
    
    /**
     * 收藏次数
     */
    @Column(name = "like_count", columnDefinition = "INTEGER DEFAULT 0")
    private Integer likeCount;
} 

File: anime-social-platform/frontend/src/types/resource.ts
/**
 * 资源数据类型
 */
export interface Resource {
  /** 资源ID */
  id: number;
  
  /** 上传用户ID */
  userId?: number;
  
  /** 上传用户名 */
  username?: string;
  
  /** 上传用户头像 */
  userAvatar?: string;
  
  /** 资源标题 */
  title: string;
  
  /** 资源描述 */
  description: string;
  
  /** 文件类型，也用作资源类型 */
  fileType?: string;
  
  /** 文件大小(KB) */
  fileSize?: number;
  
  /** 文件URL/下载链接 */
  fileUrl?: string;

  /** 封面URL */
  coverUrl?: string;
    
  /** 上传时间 */
  uploadTime: string;
  
  /** 下载次数 */
  downloadCount?: number;
  
  /** 收藏次数 */
  favoriteCount: number;
  
  /** 点赞次数 */
  likeCount: number;
  
  /** 标签列表 */
  tags: {
    id: number;
    name: string;
  }[];
  
  /** 当前用户是否已收藏该资源 */
  isFavorited?: boolean;

  /** 当前用户是否已点赞该资源 */
  isLiked?:boolean;
}

export interface ResourceListResponse {
  items: Resource[];
  total: number;
}



